src/Time/TimeStepId.hpp

```diff
@@ -64,7 +64,7 @@ class TimeStepId {
 
   int64_t slab_number_{std::numeric_limits<int64_t>::lowest()};
   Time step_time_{};
-  uint64_t substep_{0};
+  uint8_t substep_{0};
```

Comment:
Would moving this to after `substep_time_` help with reducing the class size? I assume the issue is that `Rational` is 64- or 32-bit aligned

Reply:
No.  The class size has to be a multiple of the alignment, so you still get padding even if the small field is at the end.